editor/plugins/script_text_editor.cpp

```diff
@@ -1710,6 +1710,10 @@ Variant ScriptTextEditor::get_drag_data_fw(const Point2 &p_point, Control *p_fro
 }
 
 bool ScriptTextEditor::can_drop_data_fw(const Point2 &p_point, const Variant &p_data, Control *p_from) const {
+	String file_name = script->get_path().get_file();
+	if (!file_name.contains(".gd"))
+		return false;
```

Comment:
```suggestion
	if (!file_name.contains(".gd")) {
		return false;
	}
```
But this seems like a hack that will break existing functionality such as dropping resource files, no?

Reply:
Thanks, I forgot to add the `{} `

> This may call for further abstraction so that each language can handle drag and drop the way it sees best fit

I thought about this too, but C# works differently with gdscript in things like the class scope, just drag and drop may not be enough.

Reply:
> > This may call for further abstraction so that each language can handle drag and drop the way it sees best fit
>
> I thought about this too, but C# works differently with gdscript in things like the class scope, just drag and drop may not be enough.

I think what akien meant was something like this:

```diff
class ScriptLanguage {
+    bool support_drag_and_drop();
+    void handle_drag_and_drop(int p_column, int p_line, const Ref<Resource> &p_resource);
}
```

Then, the script editor can get the script's language and check if it supports drag and drop, and then let each language handle it appropiately.

A language like C# may just return `false` if it can't support something like this, while languages like GDScript can implement their custom way of handling this if they are able to.

Reply:
Yes that's a better approach. I took a look at the current `ScriptTextEditor::drop_data_fw`, it's a little coupled with `CodeEdit` right now, but in future I suppose we can move the functionality to `ScriptLanguage` and make it more modular